edb/pgsql/delta.py

```diff
@@ -2909,6 +2909,64 @@ def delete_scalar(
             )
             ops.add_command(uncast_func)
 
+            drop_func_cache_query = textwrap.dedent(f'''
+                DO $$
+                DECLARE
+                    query_cache_row RECORD;
+                BEGIN
+                    FOR query_cache_row IN
+                        WITH
+                        types AS (
+                            SELECT
+                                pt.oid AS oid
+                            FROM
+                                pg_type pt
+                                JOIN pg_namespace pn
+                                    ON pt.typnamespace = pn.oid
+                            WHERE
+                                pn.nspname = '{old_enum_name[0]}'
+                                AND pt.typname = '{old_enum_name[1]}'
+                        ),
+                        class AS (
+                            SELECT
+                                pc.oid AS oid
+                            FROM
+                                pg_class pc
+                                JOIN pg_namespace pn
+                                    ON pc.relnamespace = pn.oid
+                            WHERE
+                                pn.nspname = 'pg_catalog'
+                                AND pc.relname = 'pg_type'
+                        ),
+                        funcs AS (
+                            SELECT
+                                substring(p.proname FROM 6)::uuid AS key
+                            FROM
+                                pg_proc p
+                                JOIN pg_depend d
+                                    ON d.objid = p.oid
+                                JOIN types t
+                                    ON d.refobjid = t.oid
+                                JOIN class c
+                                    ON d.refclassid = c.oid
+                            WHERE
+                                p.proname LIKE '__qh_%'
+                        )
+                        SELECT
+                            c.evict
+                        FROM
+                            funcs f
+                            LEFT JOIN edgedb._query_cache c
+                                ON f.key = c.key
+                        WHERE
+                            c.evict IS NOT NULL
+                    LOOP
+                        EXECUTE query_cache_row.evict;
```

Comment:
I really like that we can do this by executing the evict function

Reply:
Oh yes I should just use the `edgedb."_evict_query_cache"()` function!